contracts/src/integrations/cert/EigenDACertVerifier.sol

```diff
@@ -43,39 +49,77 @@ contract EigenDACertVerifier is
         DATypesV1.SecurityThresholds memory initSecurityThresholds,
         bytes memory initQuorumNumbersRequired
     ) {
-        if (initSecurityThresholds.confirmationThreshold <= initSecurityThresholds.adversaryThreshold) {
+        if (
+            initSecurityThresholds.confirmationThreshold <=
+            initSecurityThresholds.adversaryThreshold
+        ) {
             revert InvalidSecurityThresholds();
         }
+        if (initQuorumNumbersRequired.length == 0 || initQuorumNumbersRequired.length > 256) {
+            revert InvalidQuorumNumbersRequired(initQuorumNumbersRequired.length);
+        }
         _eigenDAThresholdRegistry = initEigenDAThresholdRegistry;
         _eigenDASignatureVerifier = initEigenDASignatureVerifier;
         _securityThresholds = initSecurityThresholds;
         _quorumNumbersRequired = initQuorumNumbersRequired;
     }
 
+    /// @notice Decodes a certificate from bytes to an EigenDACertV3
+    /// @dev This function is external for the purpose of try/catch'ing it inside checkDACert.
+    function decodeCert(
```

Comment:
dumb idea but any merit in deferring to this when decoding the cert in the clients lol?